File: Microverse Challenges/module 3/3.1 sets-Trast00/challenge.js
function findDuplicates(array) {
  // write your code here
  const set = new Set()
  const duplicate = []
  array.forEach(element => {
    if (set.has(element)) {
      duplicate.push(element)
    } else {
      set.add(element)
    }
  });
  return duplicate
}

console.log(findDuplicates([1, 2, 3, 1, 5, 6, 7, 8, 5, 2, 5]))
// => [1, 5, 2]

console.log(findDuplicates([3, 501, 17, 23, -43, 67, 5, 888, -402, 235, 77, 99, 311, 1, -43]))
// => [-43]

module.exports = findDuplicates

File: Microverse Challenges/module 5/5.2 insertion-sort-Trast00/challenge.rb
# Time : O(n^2) Space: O(1)
def sort_itself(array)
  # write your code here

  # for each element in the list (starting from the second)
  array.each_with_index do |element, index|
    next if index == 0
    # for each previous element
    for j in (index).downto(1)
      # if the previous element is greater then swap
      if array[j-1] > array[j]
        array[j], array[j-1] = array[j-1], array[j]
      end
    end
    
    puts array.join(' ')
  end
  
end

sort_itself([1, 4, 3, 6, 9, 2])
# => 1 4 3 6 9 2
#    1 3 4 6 9 2
#    1 3 4 6 9 2
#    1 3 4 6 9 2
#    1 2 3 4 6 9
